packages/coq-lsp/coq-lsp.0.2.0+8.20/opam

```diff
@@ -0,0 +1,71 @@
+synopsis: "Language Server Protocol native server for Coq"
+description:
+"""
+Language Server Protocol native server for Coq
+"""
+opam-version: "2.0"
+maintainer: "e@x80.org"
+bug-reports: "https://github.com/ejgallego/coq-lsp/issues"
+homepage: "https://github.com/ejgallego/coq-lsp"
+dev-repo: "git+https://github.com/ejgallego/coq-lsp.git"
+authors: [
+  "Emilio Jesús Gallego Arias <e@x80.org>"
+  "Ali Caglayan <alizter@gmail.com>"
+  "Shachar Itzhaky <shachari@cs.technion.ac.il>"
+  "Ramkumar Ramachandra <r@artagnon.com>"
+]
+license: "LGPL-2.1-or-later"
+doc: "https://ejgallego.github.io/coq-lsp/"
+
+depends: [
+
+  ("ocaml" {>= "5.0"} | ("ocaml" {<= "5.0"} & "memprof-limits" { >= "0.2.1" } ))
+
+  "dune"         { >= "3.13.0" } # Version interval [3.8-3.12] was
+                                 # broken for composed builds with Coq
+
+  # lsp dependencies
+  "cmdliner"     { >= "1.1.0" }
+  "yojson"       { >= "1.7.0" }
+  "uri"          { >= "4.2.0" }
+  "dune-build-info" { >= "3.2.0" }
+
+  # waterproof parser
+  "menhir"       { >= "20220210" }
+
+  # unit testing
+  "ppx_inline_test"     { >= "0.14.1" }
+
+  # Uncomment this for releases
+  "coq"          { >= "8.20" < "8.21"  }
+
+  # coq deps: remove this for releases
+  "ocamlfind" {>= "1.8.1"}
+  "zarith" {>= "1.11"}
+
+  # serlib deps: see what we need to keep for release
+  "ppx_deriving"        { >= "4.2.1"    }
+  "ppx_deriving_yojson" { >= "3.4"      }
```

Comment:
```suggestion
  "ppx_deriving_yojson" { >= "3.7"      }
```
I think you need the fix in this version to avoid
```

#=== ERROR while compiling coq-lsp.0.2.0+8.20 =================================#
# context              2.3.0~alpha~dev | linux/x86_64 | ocaml-base-compiler.5.0.0 | pinned(https://github.com/ejgallego/coq-lsp/releases/download/0.2.0%2B8.20/coq-lsp-0.2.0.8.20.tbz)
# path                 ~/.opam/5.0/.opam-switch/build/coq-lsp.0.2.0+8.20
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p coq-lsp -j 31
# exit-code            1
# env-file             ~/.opam/log/coq-lsp-7-698c6f.env
# output-file          ~/.opam/log/coq-lsp-7-698c6f.out
### output ###
# (cd _build/default && /home/opam/.opam/5.0/bin/ocamlc.opt -w -40 -g -ppx '.ppx/3e23dc93181ecfc52b6641f305d953ce/ppx.exe --as-ppx --cookie '\''library-name="petanque_json"'\''' -bin-annot -I petanque/json/.petanque_json.objs/byte -I /home/opam/.opam/5.0/lib/angstrom -I /home/opam/.opam/5.0/lib/base -I /home/opam/.opam/5.0/lib/base/base_internalhash_types -I /home/opam/.opam/5.0/lib/base/caml -I /home/opam/.opam/5.0/lib/base/shadow_stdlib -I /home/opam/.opam/5.0/lib/bigstringaf -I /home/opam/.opam/5.0/lib/biniou -I /home/opam/.opam/5.0/lib/camlp-streams -I /home/opam/.opam/5.0/lib/cmdliner -I /home/opam/.opam/5.0/lib/coq-core/boot -I /home/opam/.opam/5.0/lib/coq-core/clib -I /home/opam/.opam/5.0/lib/coq-core/config -I /home/opam/.opam/5.0/lib/coq-core/engine -I /home/opam/.opam/5.0/lib/coq-core/gramlib -I /home/opam/.opam/5.0/lib/coq-core/interp -I /home/opam/.opam/5.0/lib/coq-core/kernel -I /home/opam/.opam/5.0/lib/coq-core/lib -I /home/opam/.opam/5.0/lib/coq-core/library -I /home/opam/.opam/5.0/lib/coq-core/parsing -I /home/opam/.opam/5.0/lib/coq-core/perf -I /home/opam/.opam/5.0/lib/coq-core/pretyping -I /home/opam/.opam/5.0/lib/coq-core/printing -I /home/opam/.opam/5.0/lib/coq-core/proofs -I /home/opam/.opam/5.0/lib/coq-core/tactics -I /home/opam/.opam/5.0/lib/coq-core/vernac -I /home/opam/.opam/5.0/lib/coq-core/vm -I /home/opam/.opam/5.0/lib/easy-format -I /home/opam/.opam/5.0/lib/findlib -I /home/opam/.opam/5.0/lib/ocaml/dynlink -I /home/opam/.opam/5.0/lib/ocaml/str -I /home/opam/.opam/5.0/lib/ocaml/threads -I /home/opam/.opam/5.0/lib/ocaml/unix -I /home/opam/.opam/5.0/lib/parsexp -I /home/opam/.opam/5.0/lib/ppx_compare/runtime-lib -I /home/opam/.opam/5.0/lib/ppx_deriving/runtime -I /home/opam/.opam/5.0/lib/ppx_deriving_yojson/runtime -I /home/opam/.opam/5.0/lib/ppx_hash/runtime-lib -I /home/opam/.opam/5.0/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/5.0/lib/result -I /home/opam/.opam/5.0/lib/sexplib -I /home/opam/.opam/5.0/lib/sexplib0 -I /home/opam/.opam/5.0/lib/stringext -I /home/opam/.opam/5.0/lib/uri -I /home/opam/.opam/5.0/lib/yojson -I /home/opam/.opam/5.0/lib/zarith -I coq/.coq.objs/byte -I fleche/.fleche.objs/byte -I fleche/waterproof/.fleche_waterproof.objs/byte -I lang/.lang.objs/byte -I lsp/.lsp.objs/byte -I petanque/.petanque.objs/byte -I serlib_8_20/.serlib.objs/byte -no-alias-deps -open Petanque_json -o petanque/json/.petanque_json.objs/byte/petanque_json__JAgent.cmo -c -impl petanque/json/jAgent.ml)
# File "petanque/json/jAgent.ml", line 82, characters 4-73:
# 82 |     type t = [%import: Petanque.Agent.Premise.Info.t] [@@deriving yojson]
#          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
# Error: Unbound value Result.to_yojson
```

Reply:
Thanks! I've implemented a different fix which is to bump `ppx_import`.